cmd/alarms/main.go

```diff
@@ -0,0 +1,187 @@
+		}
+	}()
+	tracer := tp.Tracer(svcName)
+
+	dbConfig := postgres.Config{Name: defDB}
+	if err := env.ParseWithOptions(&dbConfig, env.Options{Prefix: envPrefixDB}); err != nil {
+		logger.Error(err.Error())
+	}
+
+	db, err := postgres.Setup(dbConfig, *alarmsRepo.Migration())
+	if err != nil {
+		logger.Error(err.Error())
+		exitCode = 1
+		return
+	}
+	defer db.Close()
+
+	repo := alarmsRepo.NewAlarmsRepo(db)
+
+	authConfig := grpcclient.Config{}
+	if err := env.ParseWithOptions(&authConfig, env.Options{Prefix: envPrefixAuth}); err != nil {
+		logger.Error(fmt.Sprintf("failed to load %s auth configuration : %s", svcName, err))
+		exitCode = 1
+		return
+	}
+	authn, authnClient, err := authsvc.NewAuthentication(ctx, authConfig)
+	if err != nil {
+		logger.Error(err.Error())
+		exitCode = 1
+		return
+	}
+	defer authnClient.Close()
+	logger.Info("AuthN  successfully connected to auth gRPC server " + authnClient.Secure())
+
+	domsGrpcCfg := grpcclient.Config{}
+	if err := env.ParseWithOptions(&domsGrpcCfg, env.Options{Prefix: envPrefixDomains}); err != nil {
+		logger.Error(fmt.Sprintf("failed to load domains gRPC client configuration : %s", err))
+		exitCode = 1
+		return
+	}
+
+	domAuthz, _, domainsHandler, err := domainsAuthz.NewAuthorization(ctx, domsGrpcCfg)
+	if err != nil {
+		logger.Error(err.Error())
+		exitCode = 1
+		return
+	}
+	defer domainsHandler.Close()
+
+	authz, authzHandler, err := authsvcAuthz.NewAuthorization(ctx, authConfig, domAuthz)
+	if err != nil {
+		logger.Error("failed to create authz " + err.Error())
+		exitCode = 1
+		return
+	}
+	defer authzHandler.Close()
+
+	logger.Info("AuthZ successfully connected to auth gRPC server " + authzHandler.Secure())
+
+	idp := uuid.New()
+
+	svc := alarms.NewService(idp, repo)
+
+	svc = middleware.NewAuthorizationMiddleware(svc, authz)
+	svc = middleware.NewLoggingMiddleware(logger, svc)
+	counter, latency := prometheus.MakeMetrics("alarms", "api")
+	svc = middleware.NewMetricsMiddleware(counter, latency, svc)
+	svc = middleware.NewTracingMiddleware(tracer, svc)
+
+	httpServerConfig := server.Config{Port: defSvcHTTPPort}
+	if err := env.ParseWithOptions(&httpServerConfig, env.Options{Prefix: envPrefixHTTP}); err != nil {
+		logger.Error(fmt.Sprintf("failed to load %s HTTP server configuration : %s", svcName, err))
+		exitCode = 1
+		return
+	}
+	hs := httpserver.NewServer(ctx, cancel, svcName, httpServerConfig, httpAPI.MakeHandler(svc, logger, idp, cfg.InstanceID, authn), logger)
+
+	pubSub, err := brokers.NewPubSub(ctx, cfg.BrokerURL, logger)
+	if err != nil {
+		logger.Error(fmt.Sprintf("failed to connect to message broker: %s", err))
+		exitCode = 1
+		return
+	}
+	defer pubSub.Close()
+	pubSub = brokerstracing.NewPubSub(httpServerConfig, tracer, pubSub)
+
+	consumer := consumer.NewConsumer(svc)
+	consumer = consumertracing.NewBlocking(tracer, consumer, httpServerConfig)
+	if err = consumers.Start(ctx, svcName, pubSub, consumer, cfg.ConfigPath, logger); err != nil {
```

Comment:
Do you think it might make sense to use broker.PubSub here instead of consumer?   
Since consumer relies on the transformer, my thinking is to keep the alarm independent of that.   
Ideally, the alarm would have its own format along with its own marshal and unmarshal methods.

Reply:
Consumers no longer rely on transformers - i.e. transformers can be nil - see [#2810](https://github.com/absmach/supermq/commit/a01d2571de8109f3edb80d7fc340b68493f93c0e).

Reply:
Yes, this approach would require an additional config.toml file during deployment. However, if we use a Subscriber, we could pass the subject filter through environment variables instead.